test_conformance/images/clFillImage/test_fill_generic.cpp

```diff
@@ -78,12 +78,16 @@ cl_mem create_image( cl_context context, cl_command_queue queue, BufferOwningPtr
             if (gDebugTrace)
                 log_info(" - Creating 1D buffer image %d ...\n",
                          (int)imageInfo->width);
-            if (gEnablePitch) host_ptr = malloc(imageInfo->rowPitch);
             {
                 cl_int err;
-                cl_mem buffer =
-                    clCreateBuffer(context, CL_MEM_READ_WRITE,
-                                   imageInfo->rowPitch, host_ptr, &err);
+                cl_mem_flags buffer_flags = CL_MEM_READ_WRITE;
+                if (gEnablePitch)
+                {
+                    host_ptr = malloc(imageInfo->rowPitch);
```

Comment:
As per section 5.3.1.2. Image Descriptor this host_ptr needs to be aligned with [CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT]

If the buffer object specified by mem_object was created with [CL_MEM_USE_HOST_PTR](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_MEM_USE_HOST_PTR), the host_ptr specified to [clCreateBuffer](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clCreateBuffer) or [clCreateBufferWithProperties](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clCreateBufferWithProperties) must be aligned to the maximum of the [CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT) value for all devices in the context associated with the buffer specified by mem_object that support images